### DISCUSSION TITLE
Different Time Format in Table vs Line Chart?

### DISCUSSION THREAD

**Discussion by Doctor-N0** [Budibase/budibase#11620] [Help] [answered]
I noticed that the timestamps are in a different format in tables and line charts. Is there any possibility to change to time zone to the correct format in the line chart? (The table displays the timestamp in the correct format)

Thanks so much!

![budibase](https://github.com/Budibase/budibase/assets/105864309/354d4310-beb6-4cf2-abd2-f0fd30b02dae)

**Comment by ConorWebb96** [comment]
Hey @Doctor-N0,

Your best solution in this scenario would be to use a formula column that converts the date into whatever format you desire. The graph's base functionality will convert UTC dates+time into what you are seeing.

Below is how i've solved this;
1. Create a formula column
2. Name it something relevant e.g. `formulaUpdatedAt`
<img width="100%" alt="Screenshot 2023-08-30 at 12 46 50" src="https://github.com/Budibase/budibase/assets/126772285/847d8410-1487-4638-856e-39143d3e9672">
3. Click the lighting bolt, switch over to JS
<img width="999" alt="Screenshot 2023-08-30 at 12 46 57" src="https://github.com/Budibase/budibase/assets/126772285/8eefb9e1-d4bb-4711-8db9-9c98b422838c">
4. Paste the below code in and save. (Make sure it is set to dynamic, as these values will change every time you modify the records.)

`return helpers.date($("Updated At"), "MMM DD YYYY, MM:ss")`

**Output**
<img width="1182" alt="Screenshot 2023-08-30 at 12 45 36" src="https://github.com/Budibase/budibase/assets/126772285/8808804a-6b58-4243-ad29-43eb42436d81">

I hope this helps.

**Reply by Doctor-N0** [reply]
Do you mean I should add a column called "formulaUpdatedAt" into my MySQL database, and insert the JavaScript code there?

**Reply by ConorWebb96** [reply]
Hey @Doctor-N0,

This is a Budibase specific column, you can add this to your db through the Budibase UI within the data area. It should appear as a vanilla text field within your MySQL database.
<img width="1169" alt="Screenshot 2023-08-30 at 13 01 44" src="https://github.com/Budibase/budibase/assets/126772285/7d98918d-ba90-4739-a9e7-8a88a1c46d7f">

Then set it up as I described above.

I hope this makes sense.

**Reply by Doctor-N0** [reply]
Thanks, but can I specify the timezone? Even though I run budibase on a German server, I receive some weird timestamps. I was trying something like
 `return helpers.date($("updated_at"), "MMM DD YYYY, MM:ss", "Europe/Vienna")`
but that doesn't seems to work ...

**Reply by ConorWebb96** [reply, marked as answer]
Hey @Doctor-N0,

The helper within Budibase doesn't allow you to specific a timezone in this case I guess we will need to write more complicated JS. The below should solve this issue for you.

```
function convertUTCToFormattedString(utcDateTime, timeZone) {
    const utcDate = new Date(utcDateTime);
    
    const options = {
        year: 'numeric',
        month: 'short',
        day: '2-digit',
        hour: '2-digit',
        minute: '2-digit',
        timeZoneName: 'short'
    };

    const localDate = new Date(utcDate.toLocaleString('en-US', { timeZone: timeZone }));
    const timeZoneString = localDate.toLocaleString('en-US', { timeZoneName: 'short' }).split(' ')[2];
    
    const formattedDateTime = localDate.toLocaleString('en-US', options);
    
    return formattedDateTime + ', ' + timeZoneString;
}

const utcDateTime = $("Updated At");
const timeZone = "Europe/Vienna";

const formattedDateTime = convertUTCToFormattedString(utcDateTime, timeZone);
return formattedDateTime;
```

I'd recommend amending the en-US to whatever local time zone you are in. Additionally you can alter how the date is formatted in the end, I added the returning of the time zone for testing purposes. I have tested and this seems to work. Furthermore, you may have to rework the code if you select GB or another local time zone as its `dd mm yyyy` there instead of `mm dd yyyy`.

I hope this helps.

**Reply by Doctor-N0** [reply]
nice, thanks so much! that's working
